homework.js

```diff
@@ -0,0 +1,147 @@
+Q1
+
+const summation = function(input){
+    let sum =0
+    for(let i =0;i<=input;i++){
+      sum = sum +i;
+      
+    }
+    console.log(sum);
+  }
+  
+  summation(10)
+
+  Q2
+
+  const summationEven = function(input){
+    let sum =0
+    for(let i =0;i<=input;i=i+2){
+      sum = sum +i;
+      
+    }
+    console.log(sum);
+  }
+  
+  summationEven(5)
+
+  Q3
+
+  const avg = function(arr){
+    let sum =0;
+    
+    for(let i =0;i < arr.length;i++){
+      
+      sum = sum + arr[i];
+      
+      
+    }
+    let average = sum/arr.length;
+  
+    console.log(average);
+  }
+  
+  avg([8, 2, 2, 4])
+  avg([20, 10, 0, 0])
+
+  Q4
+
+  const reverse =function(word){
+    let reWord=""
+    for(let i = word.length-1;i>0;i--){
+      reWord += word[i];
+    }
+    console.log(reWord)
+  }
+  reverse("caterpillar")
+
+  Q5
+  const addDashes = function(arr){
+    let word = ""
+    
+      word += arr.join("-")
+    console.log(word)
+    }
+    
+    
+    addDashes(['test1', 'test2', 'test3'])
+
+    Q6
+
+    const countUpAndDown = function(num){
+        let count =""
+        for(let i =1; i<=num;i++ ){
+          count += i
+          
+      
+        }
+        
+        for(let i =num-1; i >0 ;i-- ){
+          count = count + i;
+          
+        }
+        console.log(count)
+      }
+      
+      countUpAndDown(3)
+      
+      Q7
+
+      const wordsWithA = function(letterA){
+        for(let i= 0; i<letterA.length;i++){
+          if (letterA[i].search("a") == true ){
+            console.log(letterA[i]);
+          }
+        }
+      }
+      wordsWithA(['cat', 'rabbit', 'dog', 'frog'])
+
+      Q8
+      const wordsWithLetter = function(letter,word){
+  
+        for(let i= 0; i<word.length;i++){
+     
+     
+         if (word[i].search(letter) == true ){
+           console.log(word[i])
+           //I do not know why this part does not work
```

Comment:
The reason this isn't working because `.search()` returns the index of the first occurrence of the letter you're searching for, and if not found it returns `-1`.

So when it looks for 'g' in the word 'frog' it does find it in index 3 and so it will return 3 making your condition look like this `if (3==true)` which is false and that is why the condition won't apply.

 Read more about it in the [`.search()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/search#Return_value) docs. 

Since we know that the return value will be `-1` if the letter is not found, we can incorporate that in our condition
```js
 if (word[i].search(letter) !== -1 ){
```